src/osd/PG.h

```diff
@@ -526,11 +528,16 @@ class PG : DoutPrefixProvider {
     C_UpdateLastRollbackInfoTrimmedToApplied(PG *pg, epoch_t e, eversion_t v)
       : pg(pg), e(e), v(v) {}
     void finish(int) {
-      pg->lock();
+      //pg->lock();
       if (!pg->pg_has_reset_since(e)) {
-	pg->last_rollback_info_trimmed_to_applied = v;
+	if (pg->last_rollback_info_trimmed_to_applied < v) {
```

Comment:
Is the pg locked here?  Why would last_rollback_info_trimmed_to_applied be applied out of order?